gemrb/GUIScripts/pst/GUIREC.py

```diff
@@ -826,7 +826,10 @@ def AcceptLevelUp():
 	GemRB.SetPlayerStat (pc, IE_HITPOINTS, HPGained+oldhp)
 	#increase weapon proficiency if needed
 	if WeapProfType!=-1:
-		GemRB.SetPlayerStat (pc, WeapProfType, CurrWeapProf + WeapProfGained )
+		GemRB.SetPlayerStat (pc, IE_PROFICIENCYBASTARDSWORD+WeapProfType, CurrWeapProf + WeapProfGained )
```

Comment:
This one is the puzzling one. Aren't you setting one specific prof to the sum of all profs?

Reply:
Oh I see - yes that is exactly right. Each character only trains in one stat, and it all happens automatically. Otherwise I would have just plugged the whole thing into LUProfsSelection

TNO is the exception, he gains the points in the last proficiency slot, and then dialog scripts with npcs are used to actually train specific weapons (which I am pretty sure already works last time I checked)

I can see how that could all use some explanatory commenting though, it's all a bit cryptic.